setup.py

```diff
@@ -955,11 +987,73 @@ def build_torch_extension_v2(build_ext, options, torch_version):
     build_ext.build_extension(torch_mpi_lib_v2)
 
 
+def build_cmake(build_ext, ext, output_dir, options):
+    # CMake should come with pip requirement 'cmake', but let's verify
+
+    cmake_bin = 'cmake'
+    if not have_cmake:
+        # if the system doesn't have cmake before, we need to check the newly
+        # installed cmake binary is executable.
+        import cmake
+        cmake_bin = os.path.join(cmake.CMAKE_BIN_DIR, 'cmake')
+        if not os.access(cmake_bin, os.X_OK):
+            # if not executable, +x to the binary
+            os.chmod(cmake_bin, stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
```

Comment:
This is what it does to cmake:
```
root@8bf446f3dac6:/tmp# ls -l `which cmake`
---x--x--x 1 root root 5272400 Apr 24  2018 /usr/bin/cmake
```

We should make sure we *add* +x to existing attributes.